check_nightly_success/check-nightly-success/check.py

```diff
@@ -76,25 +77,42 @@ def main(
             continue
 
         latest_success[branch] = None
-        for run in sorted(branch_runs, key=lambda r: r["run_started_at"], reverse=True):
+        runs = sorted(branch_runs, key=lambda r: r["run_started_at"], reverse=True)
+        for run in runs:
             days_since_run = (now - datetime.fromisoformat(run["run_started_at"])).days
             if days_since_run > max_days_without_success:
                 break
             if run["conclusion"] in GOOD_STATUSES:
                 latest_success[branch] = run
                 break
 
+        oldest_run_valid[branch] = False
+        if len(runs) > 0:
+            run = runs[-1]
+            days_since_run = (now - datetime.fromisoformat(run["run_started_at"])).days
+            if days_since_run > max_days_without_success:
+                oldest_run_valid[branch] = True
+
     latest_branch = max(latest_success)
-    success = latest_success[latest_branch] is not None
+    has_latest_success = latest_success[latest_branch] is not None
+    success = False
 
-    if success:
+    if has_latest_success:
+        success = True
```

Comment:
Rather than setting a variable, let's just return False or True from inside the conditionals now that there are more than two arms.

Reply:
I do not like this change but I did it anyway.

Reply:
Also we are returning exit codes so I used 0/1 instead of False/True.

Reply:
Yes now that we are returning the integers directly returning 0/1 makes sense.